### PR TITLE
Apply more golint fixes

### PR DESCRIPTION
**Pull request by ibrasho**
Fixes various golint issues.

Helps in resolving: #9247 :grin: Only 33 issues remain to go.

`kubetest` introduced 30 new exported identifiers with no comments (and we have 3 other ones scattered in prow). I'll need some time to be able to add the missing comments.

### CODE REVIEW

File: prow/kube/cluster.go [hunk 1]
```diff
@@ -70,8 +68,6 @@ func GetKubernetesClient(masterURL, kubeConfig string) (kubernetes.Interface, er
 	return client, nil
 }
 
-// GetKubernetesClient retrieves the Kubernetes cluster
-// client from within the cluster
 func GetProwJobClient(masterURL, kubeConfig string) (prowjobclientset.Interface, error) {
```
Comment (idealhack) [inline prow/kube/cluster.go]: missing a comment here?

File: prow/flagutil/k8s_client.go [hunk 2]
```diff
@@ -61,12 +60,11 @@ func (o *KubernetesClientOptions) Validate(dryRun bool) error {
...
 }
 
-// Client returns a Kubernetes client.
+// KubeClient returns a Kubernetes client.
 func (o *KubernetesClientOptions) KubeClient() (kubernetes.Interface, error) {
 	return kube.GetKubernetesClient(o.masterURL, o.kubeConfig)
 }
 
-// Client returns a Kubernetes client.
 func (o *KubernetesClientOptions) ProwJobClient() (versioned.Interface, error) {
```
Comment (idealhack) [inline prow/flagutil/k8s_client.go]: missing a comment here?

File: prow/config/config.go [hunk 3]
```diff
@@ -105,7 +105,7 @@ type ProwConfig struct {
 	OwnersDirBlacklist OwnersDirBlacklist `json:"owners_dir_blacklist,omitempty"`
 
 	// Pub/Sub Subscriptions that we want to listen to
-	PubSubSubscriptions PubsubSubscriptions `json:"pubsub_subscriptions,omitempty"`
+	PubSubSubscriptions PubSubSubscriptions `json:"pubsub_subscriptions,omitempty"`
```
Comment (idealhack) [inline prow/config/config.go]: I'm not sure about this

File: prow/kube/cluster.go (hunk 1 above)
Comment (ibrasho) [reply prow/kube/cluster.go]: The old comment was wrong so I removed it until I add another one. It's updated now.

File: prow/flagutil/k8s_client.go (hunk 2 above)
Comment (ibrasho) [reply prow/flagutil/k8s_client.go]: The old comment was wrong so I removed it until I add another one. It's updated now.

File: pkg/ghclient/wrappers.go [hunk 4]
```diff
@@ -98,12 +98,12 @@ func (c *Client) CreateStatus(owner, repo, ref string, status *github.RepoStatus
...
 }
 
-type PRMungeFunc func(*github.PullRequest) error
+type mungePRFunc func(*github.PullRequest) error
 
 // ForEachPR iterates over all PRs that fit the specified criteria, calling the munge function on every PR.
 // If the munge function returns a non-nil error, ForEachPR will return immediately with a non-nil
 // error unless continueOnError is true in which case an error will be logged and the remaining PRs will be munged.
-func (c *Client) ForEachPR(owner, repo string, opts *github.PullRequestListOptions, continueOnError bool, mungePR PRMungeFunc) error {
+func (c *Client) ForEachPR(owner, repo string, opts *github.PullRequestListOptions, continueOnError bool, mungePR mungePRFunc) error {
```
Comment (fejta) [inline pkg/ghclient/wrappers.go]: Seems like a bad idea for an exported function to use a private type?

**Pull request closed without merging**